src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1376,13 +1376,23 @@ public void buildEnvironment(Run<?, ?> build, java.util.Map<String, String> env)
             }
         }
 
+        /* Check all repository URLs are not empty */
+        /* JENKINS-38608 reports an unhelpful error message when a repository URL is empty */
+        /* Throws an IllegalArgumentException because that exception is thrown by env.put() on a null argument */
+        int repoCount = 1;
+        for (UserRemoteConfig config:userRemoteConfigs) {
+            if (config.getUrl() == null) {
+                throw new IllegalArgumentException("Git repository URL " + repoCount + " is an empty string in job definition. Checkout requires a valid repository URL");
+            }
+            repoCount++;
+        }
 
         if (userRemoteConfigs.size()==1){
             env.put("GIT_URL", userRemoteConfigs.get(0).getUrl());
         } else {
             int count=1;
-            for(UserRemoteConfig config:userRemoteConfigs)   {
-                env.put("GIT_URL_"+count, config.getUrl());
+            for (UserRemoteConfig config:userRemoteConfigs) {
```

Comment:
Could you revert this portion of the change? 

Due to the large volume of pull requests in the git plugin backlog, we intentionally avoid changes that are purely whitespace.  When a whitespace change is accepted, it risks creating merge conflicts with one or more of the 60+ proposed pull requests that have the 'Later' label.